R/08-create-final-reports.R

```diff
@@ -0,0 +1,260 @@
+tmp_grad_projections <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."Graduate_Projections"'))))
+tmp_cohort_dist <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."Cohort_Program_Distributions"'))))
+
+# Exclusion Tables
+exclude_lcip_cred <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."T_Exclude_from_Projections_LCIP4_CRED"'))))
+exclude_lcp_cd <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."T_Exclude_from_Projections_LCP4_CD"'))))
+exclude_cred <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."T_Exclude_from_Projections_PSSM_Credential"'))))
+
+# Look up tables
+age_groups <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."tbl_Age_Groups"'))))
+age_groups_rollup <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."tbl_Age_Groups_Rollup"'))))
+credentials <- tibble(dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."T_PSSM_Credential_Grouping_Appendix"'))))
+
+# 1. Create Final Graduate Projections ----
+
+# back calculate grad projections from cohort distribution, excluding specific cips
+# note that for grad projections, we do not want to include PTIB 
+filtered_grads <- tmp_grad_projections %>% 
+  filter(SURVEY != 'PTIB') %>% 
+  select(YEAR, AGE_GROUP, PSSM_CRED, GRADUATES) %>% 
+  mutate(PSSM_CRED = toupper(PSSM_CRED)) %>% # include to make sure joins work correctly 
+  inner_join(
+    tmp_cohort_dist %>% 
+      filter(SURVEY != 'PTIB') %>% 
+      mutate(PSSM_CRED = toupper(PSSM_CRED)) %>% 
+      select(PSSM_CREDENTIAL, YEAR, AGE_GROUP, PSSM_CRED, LCP4_CD, GRAD_STATUS, TTRAIN, LCIP4_CRED, PERCENT),
+    by=c('YEAR', 'AGE_GROUP', 'PSSM_CRED')
+  ) %>% 
+  filter(
+    !PSSM_CREDENTIAL %in% (exclude_cred %>% pull(PSSM_CREDENTIAL)),
+    !LCP4_CD %in% (exclude_lcp_cd %>% pull(LCIP_LCP4_CD)),
+    !LCIP4_CRED %in% (exclude_lcip_cred %>% pull(LCIP4_CRED))
+  ) %>% 
+  mutate(
+    GRADS = GRADUATES*PERCENT
+  )
+
+# calculate totals by year and age group 
+grads_agg <- filtered_grads %>% 
+  mutate(PSSM_CREDENTIAL = toupper(PSSM_CREDENTIAL)) %>% # include so things aggregate correctly 
+  inner_join(
+    age_groups ,
+    by = c('AGE_GROUP' = 'AGE_GROUP_LABEL')
+  ) %>% 
+  inner_join(
+    age_groups_rollup,
+    by = 'AGE_GROUP_ROLLUP'
+  ) %>% 
+  group_by(
+    AGE_GROUP_ROLLUP_LABEL,
+    YEAR,
+    PSSM_CREDENTIAL
+  ) %>% 
+  summarize(GRADS = sum(GRADS)) %>% 
+  inner_join(credentials %>% mutate(PSSM_CREDENTIAL = toupper(PSSM_CREDENTIAL)), by='PSSM_CREDENTIAL') %>% 
+  select(
+    AGE_GROUP_ROLLUP_LABEL,
+    YEAR,
+    PSSM_CREDENTIAL_NAME,
+    GRADS
+  ) %>%
+  ungroup()
+
+# round each value to nearest 5
+grads_rounded <- grads_agg %>% 
+  mutate(GRADS = as.integer(5*round(GRADS/5, 0)))
+
+# get totals per age_group and join back in 
+grad_totals <- grads_rounded %>% 
+  group_by(AGE_GROUP_ROLLUP_LABEL, YEAR) %>% 
+  summarize(GRADS = sum(GRADS), PSSM_CREDENTIAL_NAME = 'Total')
+
```

Comment:
I think the totals are meant to be based on the grads_agg and then rounded to nearest 5 as well
```suggestion
grad_totals <- grads_agg %>% 
  group_by(AGE_GROUP_ROLLUP_LABEL, YEAR) %>% 
  summarize(GRADS = sum(GRADS), PSSM_CREDENTIAL_NAME = 'Total')  %>% 
  mutate(GRADS = as.integer(5*round(GRADS/5, 0)))
```

Reply:
Do you think so? It made sense to me for these totals to match exactly the rounded totals (and by default they will also be rounded to 5). I could see it being either way. Are there examples in the previous version where the totals don't always match the rounded totals? (If that sentence makes sense...)

Reply:
Yes, I was reviewing the previous results and most of the 45 to 64 the total was not equal to the sum, and then some scattered differences.